### ISSUE TITLE
Copter: pre-arm check "Terrain data missing" when attempting to arm without GPS lock

### ISSUE THREAD

**Issue by rmackay9** [ArduPilot/ardupilot#12913]

**Comment by rmackay9** [comment]
It's possible that this is caused by the terrain failsafe and probably only appears if the vehicle is in Auto, Guided or RTL mode while disarmed.

**Comment by peterbarker** [comment]
You're suggesting this message should be squelched, I assume :-)

**Comment by rmackay9** [comment]
Yes, this warning is misleading I think.  Saying that, I haven't seen it again myself nor has any user complained.

**Comment by IamPete1** [comment]
The cause is terrain failsafe, not related to arming directly.

https://github.com/ArduPilot/ardupilot/blob/25ee57080fb6f86487d67ad2c55f3b79e046ad67/ArduCopter/events.cpp#L270

I guess the correct fix would be to refuse to arm if were just going to instantly hit terrain failsafe.

**Comment by DanishVasdev** [comment]
Hi, I would like to work on this issue. Could you guide me on how would I reproduce this on SITL?...I am new to ardupilot so I am not sure exactly where would I find this
I tried using the "testing GPS failure" given in -
https://ardupilot.org/dev/docs/using-sitl-for-ardupilot-testing.html
But here arming with this I couldn't see any such message on the console...only "pre arm fail"  and "AP: Pre Arm: Bad Fix"

**Comment by Jai-GAY** [comment]
I got this message when testing a LUA Script. A day earlier, it was okay and I could repeatedly execute the script without any hardware or software reboot.

[CubeOrange, 4.4.4.](https://drive.google.com/file/d/1X6W0VLPWYVN5WgjdgIVD_Z5zWXTN7b5W/view?usp=drive_link)
```
GPS 1: specified as   UAVCAN1-125
Arming -> Taking off RC11 : 2
Taking off
Failsafe: Terrain data missing
Wait for arming : 2
```

**Comment by Jai-GAY** [comment]
get it again, [same 4.4.4, same script](https://drive.google.com/file/d/1kdMMJTnj2PhJRou0gc8TI0n_AndqmQ4L/view?usp=drive_link), on different day.
